### PR TITLE
Allow disabling redirects

### PR DESCRIPTION
**Pull request by escowles**
Fixes https://jira.duraspace.org/browse/FCREPO-2191

### CODE REVIEW

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        uri = create(baseUrl);
+        testBuilder = new HeadBuilder(uri, client);
+    }
+
+    @Test
+    public void testHead() throws Exception {
+        testBuilder.perform();
+
+        final ArgumentCaptor<HttpRequestBase> requestCaptor = ArgumentCaptor.forClass(HttpRequestBase.class);
```
Comment (awoods) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: These two lines seem redundant with what is in: https://github.com/fcrepo4-exts/fcrepo-java-client/pull/21/files#diff-2e64c0ef949237ed6c137fa4d26e6925R81

Any reason not to remove them?

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        verify(client).executeRequest(eq(uri), requestCaptor.capture());
+
+        final HttpRequestBase request = getRequest();
+        assertEquals(0, request.getAllHeaders().length);
```
Comment (awoods) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: What is this test verifying by asserting that there are no headers?

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java (hunk 1 above)
Comment (escowles) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: I agree — I'll remove them.

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java (hunk 2 above)
Comment (escowles) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: This was copied from the Get test, where it was in contrast to later tests that checked for certain headers to be present: https://github.com/fcrepo4-exts/fcrepo-java-client/blob/master/src/test/java/org/fcrepo/client/GetBuilderTest.java#L69-L77

In this case, maybe I should just fold the getRequest() method into the testHead() to simply verify that the request was executed?

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Test
+    public void testDisableRedirects() throws Exception {
+        testBuilder.disableRedirects();
+        assertFalse(testBuilder.request.getConfig().isRedirectsEnabled());
+    }
+
+    private HttpRequestBase getRequest() throws FcrepoOperationFailedException {
+        final ArgumentCaptor<HttpRequestBase> requestCaptor = ArgumentCaptor.forClass(HttpRequestBase.class);
```
Comment (ajs6f) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: Just as a style thing, I think it's a bit more readable to use the `@Captor` annotation.

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java (hunk 2 above)
Comment (awoods) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: That could be good.
Do you have thoughts on an integration test?
I would also suggest updating the README... but this PR is still pending: https://github.com/fcrepo4-exts/fcrepo-java-client/pull/16

Comment (escowles) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: OK, I'll update the README and I think I can add an integration test.

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java (hunk 3 above)
Comment (escowles) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: @ajs6f That could be an improvement, but none of the other tests are using it, so I went with the existing approach.  Want to make a ticket to convert the whole test suite?

**Comment by escowles** [comment]
I've pushed another commit that simplifies the Head test, updates the README, and adds an IT.

File: src/test/java/org/fcrepo/client/HeadBuilderTest.java (hunk 3 above)
Comment (ajs6f) [inline src/test/java/org/fcrepo/client/HeadBuilderTest.java]: [Have fun.](https://jira.duraspace.org/browse/FCREPO-2193)

**Comment by awoods** [comment]
:+1: looks good to me.

**Comment by escowles** [comment]
@ruebot Squashed!

**Pull request merged**